### DISCUSSION TITLE
Riverpod Invalidate and then watch again.

### DISCUSSION THREAD

**Discussion by tjbaba2** [rrousselGit/riverpod#2702] [Q&A]
I have a scenario in which i am calling a provider when user login and invalidate it when user logout. 

but when i invelidate the provider and i try to watch it again on login it doesn't work, i have to restart the app every time. 
am i doing something wrong?

**Comment by AhmedLSayed9** [comment]
What do you mean by "it doesn't work".
Also, it's hard to know the issue without a sample if your use case

**Reply by tjbaba2** [reply]
like when i login back again the watch function doesn't work, Maybe because i have invalideted the provider

**Comment by tjbaba2** [comment]
`onPressed: () async {
            Navigator.of(context, rootNavigator: true).pop('dialog');
            final sharedPrefs = ref.watch(userSessionRepositoryProvider);
            await sharedPrefs.removeUserSession();
            ref.read(firebaseAuthProvider).signOut().then((value) {
              handicapList.clear();
              ref.invalidate(homeProfileProvider);
              ref.invalidate(roundsDataProvider);
              ref.invalidate(getStatsDataProvider);
              ref.invalidate(getPlayerCoursesProvider);
              ref.invalidate(ProfileScreenProvider);
            });
            Navigator.pushAndRemoveUntil(
                context,
                MaterialPageRoute(
                    builder: (context) => const CustomSignInScreen()),
                (route) => false);
          }`

this is what i am doing on logout

**Reply by AhmedLSayed9** [reply]
This is not helpful. You need to make a minimal sample that produces the issue.
Also you should format your code using \```dart yourcode ```
